### PR TITLE
Update intellij projectfiles

### PR DESCRIPTION
**Pull request by dizzzz**
### Description:

improvement of the intellij project files. With this PR it is possible to execute tests written with the xqsuite.

### Reference:

n/a

### Type of tests:

run the xquery tests in intellij.

foo
bar

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -74,3 +74,4 @@ webapp/WEB-INF/web.xml
 webapp/xqts/config.xml
 tools/yajsw/logs/wrapper.log
 tools/yajsw/logs/wrapper.log.lck
+Project_Default.xml
```
Comment (shabanovd) [inline .gitignore]: why is it?

File: .idea/misc.xml [hunk 2]
```diff
@@ -28,7 +28,10 @@
     <option name="priority" value="Medium" />
     <option name="excludeFilter" value="" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="false" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="PDMPlugin">
```
Comment (shabanovd) [inline .idea/misc.xml]: should this be included? Is it useful anyhow?

File: eXist-db.iml [hunk 3]
```diff
@@ -52,209 +60,4 @@
     <orderEntry type="library" name="exquery-core" level="project" />
     <orderEntry type="library" name="expath" level="project" />
   </component>
-  <component name="org.twodividedbyzero.idea.findbugs">
```
Comment (shabanovd) [inline eXist-db.iml]: keep this useful plugin?

File: .gitignore (hunk 1 above)
Comment (dizzzz) [inline .gitignore]: it was added by intellij itself ; I guess this file should never be shared with users (it is in the root directory, which is highly not intellij-ish

File: .idea/misc.xml (hunk 2 above)
Comment (dizzzz) [inline .idea/misc.xml]: all changes were made by intellij..... I have no idea

File: eXist-db.iml (hunk 3 above)
Comment (dizzzz) [inline eXist-db.iml]: also here: intellij decided to get rid of it

**Comment by olvidalo** [comment]
I have found it useful to have the following in runConfigurations/Server.xml under `configuration`:

```xml
<log_file path="$PROJECT_DIR$/webapp/WEB-INF/logs/exist.log" checked="true" skipped="true" show_all="false" alias="exist.log" />
```

This will open exist.log in a tool window when starting the server from within intellij.

**Comment by dizzzz** [comment]
a good idea, I'll check....

**Comment by dizzzz** [comment]
@olvidalo I added the logfiles, thnx for the hint!
@shabanovd I don't know what to do with the plugin changes ; the current xml files are generated from a clean situation.

File: .idea/misc.xml (hunk 2 above)
Comment (dizzzz) [inline .idea/misc.xml]: I'll remove this line...

Comment (dizzzz) [inline .idea/misc.xml]: after removing and restarting intellij, this block re-appears ; i'd prefer to keep this to prevent conflicts in the future...

**Comment by dizzzz** [comment]
I think that the current state is good enough to Squash n Merge ...

File: eXist-db.iml (hunk 3 above)
Comment (shabanovd) [inline eXist-db.iml]: to keep plugin settings you need to install it

**Comment by olvidalo** [comment]
nice, thanks!

**Comment by dizzzz** [comment]
anyone? @adamretter @wolfgangmm @ljo ?

File: eXist-db.iml (hunk 3 above)
Comment (adamretter) [inline eXist-db.iml]: @dizzzz this should probably be included

**Comment by adamretter** [comment]
@dizzzz I just left one comment

**Comment by dizzzz** [comment]
@adamretter well *I* did not remove anything, *intellij* did, as I previously wrote. I guess all settings were 'default' or did someone enter customised settings?

**Comment by adamretter** [comment]
@dizzzz As @shabanovd said you should have the `FindBugs` plugin installed. I would also suggest that you should have the `PMD` plugin installed too

**Comment by dizzzz** [comment]
come on... I DO have the plugin installed, and at every intellij close, intellij REMOVES this section.
and the PMD plugin is there too, so I guess the intellij internals have changed.

**Comment by adamretter** [comment]
@dizzzz Sorry. From your explanation that was not clear to either me or @shabanovd it seems.

We can think about merging this then once the release issue is resolved

**Comment by dizzzz** [comment]
@adamretter OK, but the details were actually there. For me, at this moment, having it not pulled in (note that there is no functionality!) blocks me in development. With this PR it is finally possible to run the xquery tests in the IDE. I can't continue without risks for the contents for my future PRs :-(

**Comment by adamretter** [comment]
@dizzzz I am happy to merge this and will; Just waiting for the next release decision which we should hopefully have soon.

**Comment by dizzzz** [comment]
since this wil either end-up into whatever first 3.x.y version, and it does not toch any logic, why can't we pull-in these kind of PRs?

**Comment by shabanovd** [comment]
I can't understand that too ... everyone in agreement that next release must have bugfixes, but @adamretter just saying "no" to everything -(

**Comment by adamretter** [comment]
@shabanovd I am not saying no to everything. I have said yes to this, but I won't personally merge it right now, because from my perspective there are other issues that need to be resolved first.

PR's have never stopped people using Git from working locally, it can make things trickier but not impossible unless there are a stack of PRs (which there aren't in this instance).

At last count, excluding @dizzzz and myself, I counted 7 other core developers. I am not solely responsible for this decision. Any other core developers are able to form a consensus and merge this, I can't stop them, which is fine by me... so please stop blaming solely me.

**Comment by shabanovd** [comment]
@adamretter, understand you that or not, but other 7 core developers want to have agreement between all core developers and only you saying "no" this days

**Comment by adamretter** [comment]
@shabanovd I think you are mistaken. Decisions in the eXist community have never required a unanimous decision to be carried, otherwise it would be hard sometimes to make progress. Whilst I wouldn't say that eXist is a democracy, decisions have historically been made by majority. This is why I or anyone else as an individual can't stop things in the way you describe above.

Feel free to appeal to the other core developers

**Comment by dizzzz** [comment]
gents, this PR is not the place to discuss. Please fill in the doodle....

**Pull request merged**